src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -134,7 +134,7 @@ public Task<string> ReadAsStringAsync()
                 // BOM characters may be present even if a charset was specified.
                 if (bomLength == -1)
                 {
-                    byte[] preamble = encoding.GetPreamble();
+                    byte[] preamble = GetPreamble(encoding);
```

Comment:
I'm not understanding something about the existing code.  On the line:
```C#
encoding = encoding ?? DefaultStringEncoding;
```
we're checking if encoding is null, and if it is, we're using DefaultStringEncoding, which is UTF8.  But UTF8 is in the encoding table, so in that fallback case, we would have already checked for UTF8 and found that the UTF8 BOM didn't match (if it did match, encoding wouldn't be null and bomLength would be > 0).  Yet here we're falling back to DefaultStringEncoding and then going through the checks all over again.  What am I missing?  Shouldn't the logic instead be:
```C#
if (encoding == null)
{
    encoding = DefaultStringEncoding;
    bomLength = 0;
}
else if (bomLength == -1)
{
    ...
}
```
?